ksqldb-engine/src/main/java/io/confluent/ksql/function/udf/json/JsonConcat.java

```diff
@@ -46,6 +46,9 @@
 public class JsonConcat {
   @Udf
   public String concat(@UdfParameter final String... jsonStrings) {
+    if (jsonStrings == null) {
+      return null;
+    }
```

Comment:
Can the variadic parameter itself actually be null? I was trying to understand how the function could be invoked in this way without reflection.

Reply:
Good question.  Ksqldb does call these methods with reflection.  I think this questions boils down to "Can one construct a null array of type T?". 

I figured out how to do with it Strings.  Here's an example: `nullif('[a]', '[a]')`.

With a query like `select concat(nullif('[a]', '[a]')) from STREAM`, it returns empty instead of null.

I haven't figured out how to generate null lists/arrays for other types quite ready.  One really silly way to do would be to write a UDF which just returns a null list/array for whatever type.  

That aside, I'm hoping that the null checks are sane to keep rather than trying to reason through which functions ought not have null checks.  (Of course, if we want to dive into it, we can totally do that!:)

Reply:
I think that it makes sense to check for `null` here and below even if we can't catch an NPE via SQL right now. Given that the null check is cheap and ksqlDB is an actively developing project, IMO we should better be a bit more defensive with this kind of checks.